### PR TITLE
Handle special characters in file names.

### PR DESCRIPTION
**Pull request by awharn**
## Proposed changes

- Better handle illegal characters in USS file and data set names.

## Release Notes

Milestone: 3.6.1

Changelog:

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**All checks must pass before this pull request is reviewed by the Zowe Explorer squad.**

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`
- [ ] New ZE APIs are tested with extender types that haven't adopted yet to determine breaking changes. Can use Zowe zFTP marketplace extension.

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new unit test cases and they are passing
- [ ] I have added at least one end-to-end test for new features to validate behavior (if applicable)
- [x] I have manually tested the changes

**Deployment**

- [ ] I have tested new functionality with the FTP extension and profile verifying no extender profile type breakages introduced
- [ ] I have added developer documentation (if applicable)
- [ ] I have verified that all dependency updates (such as Zowe SDKs) in this pull request are compatible
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by anaxceron — APPROVED** [review]

File: packages/zowe-explorer/src/trees/dataset/DatasetActions.ts [hunk 1]
```diff
@@ -848,6 +848,10 @@ export class DatasetActions {
         return extensionMap;
     }
 
+    private static validatePathAndRemoveBacktracking(input: string): string {
+        return input.replace(/(?<=[\\/])\.{2}[\\/]/g, "");
+    }
```
Comment (traeok) [inline packages/zowe-explorer/src/trees/dataset/DatasetActions.ts]: - Both `DatasetActions` and `USSActions` define identical `validatePathAndRemoveBacktracking` functions, and both files already import `SharedUtils`. I'd recommend moving it there instead
- (nitpick) Technically this function does not validate the path, its just removing the backtrack components from a path. Consider renaming it to something like `stripBacktrackPathSegments`

File: packages/zowe-explorer/src/trees/uss/USSActions.ts [hunk 2]
```diff
@@ -1194,6 +1194,10 @@ export class USSActions {
         return undefined;
     }
 
+    private static validatePathAndRemoveBacktracking(input: string): string {
+        return input.replace(/(?<=[\\/])\.{2}[\\/]/g, "");
+    }
```
Comment (traeok) [inline packages/zowe-explorer/src/trees/uss/USSActions.ts]: see comment above about duplicate function implementation in `DatasetActions`

File: packages/zowe-explorer/__tests__/__unit__/trees/uss/USSActions.unit.test.ts [hunk 3]
```diff
@@ -2687,7 +2687,7 @@ describe("USS Action Unit Tests - downloading functions", () => {
             );
         });
 
-        it("should download a USS file with directory structure generation", async () => {
+        it("should download a USS file with directory structure generation 1", async () => {
```
Comment (traeok) [inline packages/zowe-explorer/__tests__/__unit__/trees/uss/USSActions.unit.test.ts]: The test names are a bit vague - "1" and "2", but no description of the cases / what's different between them. I think its good for documenting the test cases through their name (even if the description is brief), so maintainers can ensure that any future changes to these tests or code continue to cover the intended cases.

Rename to describe behavior ("removes parent-directory segments before building the local path") or collapse into `it.each` tables.

**Review by traeok — DISMISSED** [review]
Thanks for working on this Andrew! I left a few suggestions and I'm requesting changes regarding the duplicate function implementation.

Also, not to open a can of worms, but I believe `\` is a valid USS character for filenames. I plan to do some more manual testing around that very-rare edge case, but everything is working as expected ATM for typical filenames.

File: packages/zowe-explorer/src/trees/dataset/DatasetActions.ts (hunk 1 above)
Comment (traeok) [reply packages/zowe-explorer/src/trees/dataset/DatasetActions.ts]: De-duplicated the path helper and renamed it in 099adac

File: packages/zowe-explorer/src/trees/uss/USSActions.ts (hunk 2 above)
Comment (traeok) [reply packages/zowe-explorer/src/trees/uss/USSActions.ts]: 099adac

File: packages/zowe-explorer/__tests__/__unit__/trees/uss/USSActions.unit.test.ts (hunk 3 above)
Comment (traeok) [reply packages/zowe-explorer/__tests__/__unit__/trees/uss/USSActions.unit.test.ts]: Added brief descriptions to unit cases in 8f8fe0c

**Review by CBforZ — APPROVED** [review]
LGTM

**Review by t1m0thyj — APPROVED** [review]
